drivers/serial/uart_numicro.c

```diff
@@ -8,13 +8,19 @@
 #include <zephyr/drivers/uart.h>
 #include <NuMicro.h>
 #include <string.h>
+#ifdef CONFIG_PINCTRL
+#include <zephyr/drivers/pinctrl.h>
+#endif
```

Comment:
is pinctrl really optional in these drivers? what other method you have to configure pins? (remember we are about to deprecate pinmux/other legacy methods in 3.2)

Reply:
ack, I removed the ifdefs